tools/state-viewer/src/commands.rs

```diff
@@ -1057,6 +1062,147 @@ pub(crate) fn clear_cache(store: Store) {
+
+        let Some(state_record) = state_record else { return None::<StateRecord>.fmt(f) };
+
+        f.debug_struct("StateStatsStateRecord")
+            .field("account_id", &state_record_to_account_id(&state_record).as_str())
+            .field("type", &state_record.get_type_string())
+            .field("size", &self.size())
+            .finish()
+    }
+}
+
+#[derive(Default)]
+pub struct StateStats {
+    pub total_size: ByteSize,
+    pub total_count: usize,
+
+    pub top_state_records: BinaryHeap<StateStatsStateRecord>,
+
+    pub middle_state_record: Option<StateStatsStateRecord>,
+}
+
+impl core::fmt::Debug for StateStats {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        let average_size = self
+            .total_size
+            .as_u64()
+            .checked_div(self.total_count as u64)
+            .map(ByteSize::b)
+            .unwrap_or_default();
+        f.debug_struct("StateStats")
+            .field("total_size", &self.total_size)
+            .field("total_count", &self.total_count)
+            .field("average_size", &average_size)
+            .field("middle_state_record", &self.middle_state_record.as_ref().unwrap())
+            .field("top_state_records", &self.top_state_records)
+            .finish()
+    }
+}
+
+impl StateStats {
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    pub fn push(&mut self, key: Vec<u8>, value: Vec<u8>) -> () {
+        self.total_size += key.len() as u64 + value.len() as u64;
+        self.total_count += 1;
+
+        self.top_state_records.push(StateStatsStateRecord { key, value });
+        if self.top_state_records.len() > 5 {
+            self.top_state_records.pop();
+        }
+    }
+}
+
+fn read_flat_state_value(
+    trie_storage: &TrieDBStorage,
+    flat_state_value: FlatStateValue,
+) -> Vec<u8> {
+    match flat_state_value {
+        FlatStateValue::Ref(val) => trie_storage.retrieve_raw_bytes(&val.hash).unwrap().to_vec(),
+        FlatStateValue::Inlined(val) => val,
+    }
+}
+
+pub(crate) fn print_state_stats(home_dir: &Path, store: Store, near_config: NearConfig) {
+    let (epoch_manager, runtime, _, block_header) =
+        load_trie(store.clone(), home_dir, &near_config);
+
+    let block_hash = *block_header.hash();
+    let shard_layout = epoch_manager.get_shard_layout_from_prev_block(&block_hash).unwrap();
+
+    for shard_uid in shard_layout.get_shard_uids() {
+        let flat_storage_manager = runtime.get_flat_storage_manager();
+        flat_storage_manager.create_flat_storage_for_shard(shard_uid).unwrap();
+        let trie_storage = TrieDBStorage::new(store.clone(), shard_uid);
+        let chunk_view = flat_storage_manager.chunk_view(shard_uid, block_hash).unwrap();
+
+        let mut state_stats = StateStats::new();
+
+        let iter = chunk_view.iter_flat_state_entries(None, None);
+        for item in iter {
+            let Ok((key, value)) = item else {
```

Comment:
I like this syntax! Didn't know it existed.

Reply:
nit: Looking at the implementation of iter_flat_state_entries, it might be fine to just unwrap key value here but can leave as is.

Reply:
The sytax is cool yeah, just learned it recently too. :) 
Actually I wanted to print a warning in there, let me do that.